BaragonService/src/main/java/com/hubspot/baragon/service/elb/ApplicationLoadBalancer.java

```diff
@@ -29,23 +62,401 @@ public ApplicationLoadBalancer(Optional<ElbConfiguration> configuration,
     this.elbClient = elbClient;
   }
 
-  public boolean isInstanceHealthy(String instanceId, String name) {
-    return true;
-  }
+  @Override
+  public boolean isInstanceHealthy(String instanceId, String elbName) {
+    DescribeTargetHealthRequest healthRequest = new DescribeTargetHealthRequest()
+        .withTargets(new TargetDescription().withId(instanceId));
+    DescribeTargetHealthResult result = elbClient.describeTargetHealth(healthRequest);
 
-  public Optional<LoadBalancer> getElb(String elbName) {
-    return Optional.absent();
+    for (TargetHealthDescription health: result.getTargetHealthDescriptions()) {
+      if (health.getTargetHealth().getState().equals(TargetHealthStateEnum.Healthy.toString())
+          && health.getTarget().getId().equals(instanceId)) {
+        return true;
+      }
+    }
+
+    return false;
   }
 
+  @Override
   public void removeInstance(Instance instance, String elbName, String agentId) {
-
+    /* TODO: Need to map v1 Instance -> v2 Target */
   }
 
+  @Override
   public RegisterInstanceResult registerInstance(Instance instance, String elbName, BaragonAgentMetadata agent) {
+    /* TODO: Need to map v1 Instance -> v2 Target */
     return RegisterInstanceResult.ELB_AND_VPC_FOUND;
   }
 
-  public void syncAll(Collection<BaragonGroup> groups) {
+  @Override
+  public void syncAll(Collection<BaragonGroup> baragonGroups) {
+    try {
+      for (BaragonGroup baragonGroup : baragonGroups) {
+        if (baragonGroup.getSources().isEmpty()) {
+          LOG.debug("No traffic sources present for group {}", baragonGroup.getName());
+        } else {
+          Collection<LoadBalancer> elbsForBaragonGroup = getLoadBalancersByBaragonGroup(baragonGroup);
```

Comment:
we've had problems in the past with hitting aws' api rate limits. Rather than making an individual call for each (even though it is filtered down), it might be wise to move this outside of the loop. We can describe all ALBs in one call, then grab what we need from the object. It's likely faster to loop through the object in memory than to make a new call for new data.